cask/src/cask/router/Decorators.scala

```diff
@@ -14,10 +15,10 @@ import cask.model.{Request, Response}
   * to `wrapFunction`, which takes a `Map` representing any additional argument
   * lists (if any).
   */
-trait Decorator[OuterReturned, InnerReturned, Input] extends scala.annotation.Annotation {
+trait Decorator[OuterReturned, InnerReturned, Input, InputContext] extends scala.annotation.Annotation {
```

Comment:
this is the main change here, along with adding an extra parameter to `delegate` and changing invocation to accommodate these changes